### ISSUE TITLE
Unable to access API?

### ISSUE THREAD

**Issue by dwightmulcahy** [netmanchris/pyuhooair#3]
just testing out this and when I do this
```
auth = UhooAuth(email, password)
devices = get_all_devices(auth)
```

`devices` returns this:
```
{'status': 2, 'message': 'You are not authorized to access this API.'}
```

guessing that you have to have an Enterprise account to access the rest API... :/

**Comment by anomaly256** [comment]
FYI, I was told by a uHoo rep back in August (possibly unofficially though so don't bet the farm on it) that they intend to release a set of APIs for home users to create custom integrations at the end of the year.

I've emailed them just now asking if there's any news or information available about it.

**Comment by netmanchris** [comment]
Yes - this was built in a business account which is required for API access. 

I’d love to see them release a consumer API. It would be great to get this into Home Assistant. 

Christopher Young
@netmanchris/ Twitter

**Comment by dwightmulcahy** [comment]
this is what I got when asked about a consumer API...

> ...I will feedback this to our engineering team. We're continuously gathering feedback on features that users would like to have. Thus far, **requests for API access have been very few** but we will continue to gather feedback and if that increases, we will consider building REST APIs for users to access. 

...so if you are interested in a consumer REST API you ought to email their support so that they know there **_really_** is a demand for it.

Here is the link: https://uhooair.com/contact/

**Comment by anomaly256** [comment]
Yeah this is the response to my email this week, slightly different tune to before:

> APIs for home users are part of our roadmap but we do not have an ETA for it yet. It depends on the amount of requests we get. Thus far, there has only been a small handful of users who have requested. As we get more requests, we may consider pushing the ETA forward.

So... make some noise I Guess?  Kind of absurd they don't do this given the premium price they charge for the device itself imo

**Comment by dwightmulcahy** [comment]
A couple of friends also have the Uhoo (by my suggestion) and they voiced their interest in having an API with them too.  Pretty much got the same canned response back.

>Kind of absurd they don't do this given the premium price they charge for the device itself

+1 on that.

**Comment by anomaly256** [comment]
I decided to ask about API acces for home users again last week, got this response:

`Thanks for your message. APIs for home users and Home Assistant integration will be available in the second quarter of this year. The team has already moved this up in priority as a number of users have started requesting for it in the last 2-3 months. 
`
🤞

**Comment by dwightmulcahy** [comment]
awesome!  I'll definitely be on that when they allow access.

_**as a number of users have started requesting for it in the last 2-3 months.**_

hmmmm...  nice work guys!!!

dWiGhT

**Comment by dwightmulcahy** [comment]
still no API... :(

**Comment by matteofranceschini** [comment]
I wrote them yesterday, but I'm not sure how to interpret it...

_We are planning to release a uHoo Premium product in September where one of the features is integration with HomeAssistant. The other features would be deeper analytics on your data, comparing different rooms and sensors against each other, and access to a web portal, to name a few._

**Comment by matteofranceschini** [comment]
I've asked for clarification, here it is:

_Thanks for your message. It is compatible with your existing uHoo. uHoo Premium is an in-app subscription to upgrade your account to a uHoo Premium account to gain access to those additional features. You don't need to change your existing uHoo._

**Comment by dwightmulcahy** [comment]
>uHoo Premium is an in-app subscription

So $$/month... 🤔

was hoping I didn't have to spend time reverse engineering the mobile app...

**Comment by matteofranceschini** [comment]
I'm not so into reversing iOS apps, but the hardware is quite standard.. By accepting to lose their calibration (if any), it's probably not so difficult to write a custom firmware. Sadly, an ESP32/ESP8266 would've be waaaay simpler (for me) to handle.

Cool thing, someone already opened it:
https://wiki.liutyi.info/display/CO2/uHoo+inside

**Comment by jererobles** [comment]
You can look at my implementation here https://github.com/jererobles/airq/blob/master/airq/api/crypto.py

**Comment by jererobles** [comment]
If somebody wants to port that to Node.js so I can use with Homebridge I'd be super happy 😊

**Comment by iqbalhs** [comment]
any updates about this?
is uhoo premium released yet?

**Comment by MilitaryNerd** [comment]
Now it seems obvious why the Home Assistant HACS integration is broke. Is the above code to work with the Uhoo device directly?

**Comment by anomaly256** [comment]
> > uHoo Premium is an in-app subscription
> 
> So $$/month... 🤔

Worth noting they've just launched a home user version of 'premium' with web dashboard, integrations etc.  Free for current owners for 1 year, $10/month (I assume USD) after that which is still more accessible than the 'office' premium plans previously.

The app also highlights a Home Assistant integration 'coming soon'.

Not an advertisement, just noting this is probably the user API they mentioned to me 2 years ago finally being delivered.

Edit: can confirm the premium web dashboard is now accessible on this plan, with API key retrievable from user settings and json data sent to client.

![178183732-d0955249-589b-43cd-933e-d37db07758a5](https://user-images.githubusercontent.com/421854/178184125-5884b28d-435b-4e09-b3f8-60d4e3298d84.png)![178183746-acf01018-fcea-4557-b3a9-fc79bc8e1a02](https://user-images.githubusercontent.com/421854/178184119-55d3a326-e35a-42f1-b7bf-4c9f20f8fc35.png)
